### PR TITLE
PLT-97 Add style guide and a link to it from CONTRIBUTING doc.

### PR DESCRIPTION
**Pull request by thealmarty**
- I commented out a couple things that we aren't very sure about. Changed the wording a bit in a couple places. 
- Added a link under the "Working conventions" section, which maybe could use a tidy up now that we have the style guide. It's also OK as is, it's not very repetitive. 

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: CONTRIBUTING.adoc
Comment (michaelpj) [inline CONTRIBUTING.adoc:172]:
```suggestion
Please follow our link:STYLEGUIDE{outfilesuffix}[Haskell style guide], which documents most of our conventions for working on Haskell code.
```

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:6]: bullet points got lost?

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:17]: these parens are silly actually
```suggestion
Use 4 spaces for indentation. Avoid lines over 120 characters. This is codified in `.editorconfig`.
```

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:1]: add a TOC with `:toc:`?

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:39]: This seems to format a bit weirdly. Probably these should just be lower-level headings?

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:61]: Asciidoc has source blocks with specific languages, which in principle should get us syntax higlighting: https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:231]:
```suggestion

=== When to use hierarchical re-exporting modules
```

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:233]: I don't generally think it's worth including such TODOs, I'm not sure it makes us any more likely to remember to write something.

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:249]: semantically these seem like unordered lists, but not a big deal

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:298]: just delete it

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:354]: delete

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:379]: format with some verbatim formatting

**Review by michaelpj — COMMENTED** [review]
A few comments. Worth looking at the rich diff on Github, makes rendering issues more obvious!

**Comment by thealmarty** [comment]
> A few comments. Worth looking at the rich diff on Github, makes rendering issues more obvious!

Sorry I was using the vscode preview function, maybe that doesn't render the same as on here. I'll remember that for next time.

**Comment by michaelpj** [comment]
> Sorry I was using the vscode preview function, maybe that doesn't render the same as on here. I'll remember that for next time.

Huh, weird? I think asciidoc isn't fully standardized, so maybe that's it...

File: STYLEGUIDE.adoc
Comment (thealmarty) [reply STYLEGUIDE.adoc:39]: Actually this is rendering right on github it seems (just like in the google doc). But yeah I can make them headings instead.

Comment (michaelpj) [reply STYLEGUIDE.adoc:39]: I get strange stuff where it makes weird collapsible sections or something?

**Comment by thealmarty** [comment]
I think I got everything? Ready to merge once approved.

File: STYLEGUIDE.adoc
Comment (michaelpj) [inline STYLEGUIDE.adoc:418]: These aren't haskell! So it highlights weirdly.

**Review by michaelpj — APPROVED** [review]
Just one tiny thing, then feel free to merge.

**Pull request merged**
